### PR TITLE
feat(snuba): Respect Organization retention in queries

### PR DESCRIPTION
**Pull request by bretthoerner**
Fixes SNS-125

### CODE REVIEW

File: src/sentry/utils/snuba.py [hunk 1]
```diff
@@ -96,6 +101,16 @@ def raw_query(start, end, groupby=None, conditions=None, filter_keys=None,
...
 
+    # any project will do, as they should all be from the same organization
+    project = Project.objects.get(pk=project_ids[0])
+    retention = quotas.get_event_retention(
+        organization=Organization(project.organization_id)
+    )
+    if retention:
+        start = max(start, datetime.utcnow() - timedelta(days=retention))
+        if start >= end:
+            return {}
```
Comment (bretthoerner) [inline src/sentry/utils/snuba.py]: Is it safe to return an empty dict? Do any callers expect certain keys to exist even if there are no results?

File: tests/snuba/test_snuba.py [hunk 2]
```diff
@@ -122,18 +122,39 @@ def _query_for_issue(group_id):
...
+                start=base_time - timedelta(days=90),
+                end=base_time + timedelta(days=1),
+                groupby=['project_id'],
+                filter_keys={
+                    'project_id': [self.project.id],
+                },
+            )
+
+        assert _get_event_count() == {self.project.id: 2}
+        with self.options({'system.event-retention-days': 1}):
```
Comment (bretthoerner) [inline tests/snuba/test_snuba.py]: Using what's available in open source, but note this is richer in prod: https://github.com/getsentry/getsentry/blob/6043f6d5e75fadc426e3bec9dfea4ba80ab27b25/getsentry/quotas.py#L194-L202

**Review by alex-hofsteede — COMMENTED** [review]
Looks good, just the one comment about the error result

**Comment by bretthoerner** [comment]
> Maybe even raise a SnubaError for a query outside the retention bounds

I thought about that, but it would mean every caller would have to implement the same logic that I did here (fetching retention and trimming the dates). That seemed unnecessarily burdensome but I could be wrong?

**Review by alex-hofsteede — APPROVED** [review]

**Pull request merged**